spec/amp-html-layout.md

```diff
@@ -23,226 +23,180 @@ so that the runtime is able to infer sizing of elements before any remote resour
 JavaScript and data calls, have been completed. This is important since this significantly
 reduces rendering and scrolling jank.
 
-With this in mind the AMP Layout System is designed to support few but flexible layouts
+With this in mind, the AMP Layout System is designed to support few but flexible layouts
 that provide good performance guarantees. This system relies on a set of attributes such
 as `layout`, `width`, `height`, `sizes` and `heights` to express the element's layout and
 sizing needs.
 
+## Behavior
 
-## Layout Types
+A non-container AMP element (i.e., `layout != container`) starts up in the unresolved/unbuilt mode in which 
+all of its children are hidden except for a placeholder (see `placeholder` attribute). The JavaScript
+and data payload necessary to fully construct the element may still be downloading and initializing,
+but the AMP runtime already knows how to size and lay out the element only relying on CSS classes and
+`layout`, `width`, `height` and `media` attributes. In most cases, a `placeholder`, if specified, is
+sized and positioned to take all of the element's space.
 
-AMP provides a set of layouts that AMP elements can use. They are defined using layout
-attributes described below. Not every element supports every layout - you have to refer
-to the element's documentation to know which layouts you can use.
+The `placeholder` is hidden as soon as the element is built and its first layout complete. At this
+point, the element is expected to have all of its children properly built and positioned and ready
+to be displayed and to accept a reader's input. This is the default behavior. Each element can override
+to, e.g., hide `placeholder` faster or keep it around longer.
 
-### `nodisplay`
+The element is sized and displayed based on the `layout`, `width`, `height` and `media` attributes
+by the runtime. All of the layout rules are implemented via CSS internally. The element is said to
+"define size" if its size is inferable via CSS styles and does not change based on its children:
+available immediately or inserted dynamically. This does not mean that this element's size cannot
+change. The layout could be fully responsive as is the case with `responsive`, `fixed-height`, `fill` and
+`flex-item` layouts. It simply means that the size does not change without an explicit user action, e.g.
+during rendering or scrolling or post download.
 
-The element is not displayed. This layout can be applied to every AMP element. It assumes
-that the element can display itself on user action, e.g. `amp-lightbox`.
+If the element has been configured incorrectly, it will not be rendered at all in PROD and in DEV mode; the runtime will display the element in the error state. Possible errors include invalid or unsupported
```

Comment:
These are two separate paths:

- In PROD: the element will not render
- In DEV: the element will render in error state.